src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.Netstandard2_0.cs

```diff
@@ -0,0 +1,837 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.DependencyModel
+{
+    public class DependencyContextJsonReader : IDependencyContextReader
+    {
+        private readonly IDictionary<string, string> _stringPool = new Dictionary<string, string>();
+
+        public DependencyContext Read(Stream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
+            }
+
+            ArraySegment<byte> drained = ReadToEnd(stream);
+
+            try
+            {
+                return ReadCore(drained.Array);
+            }
+            catch
+            {
+                // Holds document content, clear it before returning it.
+                drained.AsSpan().Clear();
+                ArrayPool<byte>.Shared.Return(drained.Array);
+                throw;
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                _stringPool.Clear();
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+        }
+
+        private const int UnseekableStreamInitialRentSize = 4096;
+
+        private static ArraySegment<byte> ReadToEnd(Stream stream)
```

Comment:
Borrowed from https://github.com/dotnet/corefx/blob/d1da417dbf38079f06b397bdf3c8f51b94e62238/src/System.Text.Json/src/System/Text/Json/JsonDocument.Parse.cs#L182. I am working on providing better stream support with the reader, but it still results in the calling code being overly complex.